pkg/reconciler/v1alpha1/autoscaling/kpa_scaler.go

```diff
@@ -95,21 +91,11 @@ func (rs *kpaScaler) Scale(kpa *kpa.PodAutoscaler, desiredScale int32) error {
 		return err
 	}
 
-	currentScale := scl.Spec.Replicas
-	if desiredScale == currentScale {
-		return nil
-	}
-
-	// Don't scale if current scale is zero. Rely on the activator to scale
-	// from zero.
-	if currentScale == 0 {
-		logger.Info("Cannot scale: Current scale is 0; activator must scale from 0.")
-		return nil
-	}
-	logger.Infof("Scaling from %d to %d", currentScale, desiredScale)
-
-	// When scaling to zero, flip the revision's ServingState to Reserve.
-	if desiredScale == 0 {
+	// When scaling to zero, flip the revision's ServingState to Reserve,
+	// and when that propagates back to us, scale the target reference to zero.
```

Comment:
Nit: this doesn't quite make sense with your changes.  We are now setting the desired scale to zero.  Then updating the serving state when it propagates back us.  Not the other way around.  Am I reading that right?

Reply:
(Added a hold in case you want to update this comment.)

Reply:
Pushed an update, LMK if that clarifies things